### DISCUSSION TITLE
TypeScript SDK for Morphir-Elm

### DISCUSSION THREAD

**Discussion by ghost** [finos/morphir#59] [General]
The Morphir-Elm Scala backend expects some precreated types in `Morphir.SDK` namespace to exist. 

The Scala SDK exists in [morphir-jvm repo](https://github.com/finos/morphir-jvm), which is cloned every time we run `npm run build` and copied to `redistributable/`. The `morphir-elm gen` command has a `copyRedistributables()` function which seems to then include this in the generated code.

The TypeScript backend also needs an externally defined type: `Morphir.SDK.Char`, which doesn't correspond to any builtin TypeScript type. We can implement this, but how do we store and distribute this code? I see several options:

  * Inside morphir-elm repo
  * In a new Morphir-TypeScript repo, which we clone during `npm build`
  * Inside a new repo, but fetched using Git submodules or some other system
  * Somewhere else

**Comment by AttilaMihaly** [comment]
Let's go with the quickest solution for now. If the only reason we would need this for now is `Char` we can simply map that to `string` in which case we don't need a TypeScript SDK at all.

When we get to generating actual business logic we will probably need the SDK and then we can make a decision on this.

**Reply by ghost** [reply]
Fine by me, thanks for the quick response!
